deployment/stacks/api.ts

```diff
@@ -0,0 +1,423 @@
+import { Construct } from "constructs";
+import {
+  aws_apigateway as apigateway,
+  aws_ec2 as ec2,
+  aws_events as events,
+  aws_events_targets as targets,
+  aws_iam as iam,
+  aws_logs as logs,
+  aws_s3 as s3,
+  aws_s3_notifications as s3notifications,
+  aws_wafv2 as wafv2,
+  Duration,
+  RemovalPolicy,
+  Tags,
+} from "aws-cdk-lib";
+import { Lambda } from "../constructs/lambda";
+import { WafConstruct } from "../constructs/waf";
+import { getSubnets } from "../utils/vpc";
+import { LambdaDynamoEventSource } from "../constructs/lambda-dynamo-event";
+import { DynamoDBTableIdentifiers } from "../constructs/dynamodb-table";
+import { isDefined } from "../utils/misc";
+import { isLocalStack } from "../local/util";
+
+// TODO: does this need to point to the tsconfig.json file in services/app-api?
+
+interface CreateApiComponentsProps {
+  scope: Construct;
+  stage: string;
+  project: string;
+  isDev: boolean;
+  vpc: ec2.IVpc;
+  kafkaAuthorizedSubnets: ec2.ISubnet[];
+  tables: DynamoDBTableIdentifiers[];
+  brokerString: string;
+  iamPermissionsBoundary: iam.IManagedPolicy;
+  iamPath: string;
+  wpFormBucket: s3.IBucket;
+  sarFormBucket: s3.IBucket;
+  templateBucket: s3.IBucket | undefined;
```

Comment:
What circumstances would have this undefined?

Reply:
Running locally because the uploads components are skipped then because lambda layers are not available to us in localstack community edition.